src/components/SummaryManagement/summaryGroupTableBody.jsx

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import React, { useState, useEffect } from 'react';
+import './summarygroup.css';
+import { DELETE, EDIT, VIEW_SUMMARY } from '../../languages/en/ui';
+import { updateBadge } from 'actions/badgeManagement';
+
+const summaryGroupTableBody = props => {
+  const [seeReports, setSeeReports] = useState(false);
+  const [seeDelete, setSeeDelete] = useState(true);
+  const [seeEdit, setSeeEdit] = useState(true);
+  const [isReceiver, setIsReceiver] = useState(true);
+
+  useEffect(() => {
+    const roles = props.currentUserRole;
+    updateinfo(roles);
+    if (roles) {
+      if (roles === 'Manager' || roles === 'Mentor') {
+        setSeeReports(true);
+        setSeeDelete(false);
+        setSeeEdit(false);
+      }
+    }
+  }, [props.currentUserRole]);
+
+  const isASummaryReceiver = async groupId => {
+    const id = props.currentUserId;
+    const groupSummaryReceivers = await props.fetchSummaryReceiver(groupId);
+    if (id) {
+      const isSummaryReceiver = await groupSummaryReceivers.summaryReceivers;
+      const isReceiver = isSummaryReceiver.some(item => item._id === id);
+      setIsReceiver(isReceiver);
+      return isReceiver;
+    }
+    return false;
+  };
+
+  const updateinfo = async roles => {
+    const id = props.summaryGroupId;
+    if (id && !roles) {
+      await props.updateUserInfo(id);
+    }
+  };
+  if (isReceiver || props.currentUserRole === 'Administrator') {
```

Comment:
> Hey @OAkinbode, I have retested this PR. The 404 error has been solved! But there is a new issue. The list of groups didn't show up. In the video, I created a group and got 200 from the backend, which means creating successfully! But after I refreshed the page, it didn't display.
> 
>  Screen.Recording.2023-07-28.at.14.44.46.mov

Hey @OAkinbode, I found what my issue is. I tested the PR with Owner account. But this line of code only makes it available to the admin. That's why I cannot see the table content. But this also confused me as to why an owner can not see the list but is allowed to create it. Please correct me if I am wrong!

Also, a reminder can be added here if the user doesn't have access to the table.

Reply:
Hello @sxiong5. Thank you for the review. The way it works at the moment, only an admin can fully create/approve a group at the initial stage. A manager/mentor might be able to create the group, but they can have access only after an admin adds them as the summary reviewer. After that, they could add other members as needed. The intention is to control the creation of groups, but allow for managers/mentors to have some input. That workflow may be changed at some point in the future as Jae and other managers see fit.

Reply:
Yes, I agree. But should the Owner and Admin have the same permission? My issue is that I cannot access the group list as an Owner (which should have the highest permission).

Reply:
Yes, I agree. Owner should have same permission as Admin. Please update to include that.